src/game_state/GameState.cpp

```diff
@@ -147,20 +147,29 @@ void GameState::SetPlayingState(PlayingState playing_state) {
     case GameState::PlayingState::PLAYING:
       // on resume start the music back up
       if (music->getStatus() != sf::SoundSource::Status::Playing) {
+        if (previously_paused) {
+          effects.Unpause();
+          while (!effects.CommingBackFromAPause())
+            ;  // coming in from a pause
```

Comment:
I love this but it would probably be good to have some visual feedback that you pressed escape and the game is unpausing in the future